Documentation/contributors/design-proposals/installed-api.md

```diff
@@ -0,0 +1,282 @@
+
+Due to these factors, an additional means of providing this information is required.
+
+## Proposal
+
+Define a new read-only API resource, `installed.packages.operators.coreos.com`, that surfaces metadata about an operator installed and configured to watch a namespace. Back that API by an extension api-server that _synthesizes_ those resources on request.
+
+### Naming
+
+The resource will share a name with the `CSV` of the installed operator it's tracking.
+
+### Namespaces
+
+The resource will be namespaced. Additionaly, it will exist only when an operator that provides a package matching its `metadata.name` is installed and configured to watch its namespace.
+
+### Spec and Status
+
+The resource __will not__ define a `spec` field because it's intended to aggregate the state of existing resources. In other words, there is no meaning in declaring a _desired state_. However, It __will__ define a `status` field that embeds the latest `CSV` and `Subscription`. Embedding entire resources instead of their spec or status directly allows tracked resources to be added or removed more easily in subsequent API changes.
+
+### Labels and Annotations
+
+Labels of the format `operators.coreos.com/<short>: <name>` will be added for each resource tracked in the `status` (see the [k8s docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) for the inspiration behind the key prefix). For tracking `CSV`s and `Subscription`s, the label keys would include `operators.coreos.com/csv` and `operators.coreos.com/sub`.
+
+Labels on `Installed` resources must provide enough information to be queriable by the [openshift console](https://github.com/openshift/console). As such, labels may need to be projected from tracked resources or added as needed.
+
+### The `Installed` Resource
+
+Example `Installed` resource for the `etcdoperator.v0.9.4-clusterwide` `CSV`:
+
+```yaml
+apiVersion: v1alpha2/packages.operators.coreos.com
+kind: Installed
+metadata:
+  labels:
+    # Names of related resources
+    operators.coreos.com/sub: etcd
+    operators.coreos.com/csv: etcdoperator.v0.9.4-clusterwide
+    # TODO: Project labels from CSV/Subscription?
+  # Named after the package the installed operator belongs to
+  name: etcdoperator.v0.9.4-clusterwide
+  namespace: local
+status:
+    # Entire Subscription responsible for installation
+    subscription:
```

Comment:
I think it’s worth exploring a variant that doesn’t have the objects nested, but just has the fields we want. 

The way you have here could be a. Easier transition for the UI, though

Reply:
That's fair. The reason I went with nesting is because I thought it would provide an easier path when our APIs become unified - though I could be wrong.

Reply:
Since we've been talking about moving away from / combining the two, so I thought not having the names there would make it simpler?